libs/frontend/ui-build/src/components/component-menu/ComponentEventEdit.vue

```diff
@@ -207,22 +219,23 @@ const updateEventName = (name: ComponentEventType | undefined) => {
     if (!newEvent.value.eventParams?.interval) {
       newEvent.value.eventParams = {
         interval: {
+          type: ComponentArgPrimitive.Number,
           value: '1000',
         },
       }
     }
   } else if (name === ComponentEventType.ScrollIntoView) {
-    if (newEvent.value.eventParams?.interval === undefined) {
```

Comment:
I removed this if statement to make the behavior feel more natural when switching between different event types.
I think the if statement is here for preserving event params when event type changes, but that also prevents the params for scrollIntoView event to be added.

https://github.com/samatechtw/pubstudio-builder/assets/59676941/72ce7d73-3d7a-46d5-bbc5-adc2071713e1

One way to preserve the event params for all event types individually might be to include the event type in the params, but that kind of clashes with our existing data structure and might take some effort to refactor.

Reply:
This change makes sense to me! I don't think preserving the event params is very important.